pkg/apis/networking/types.go

```diff
@@ -336,19 +340,58 @@ type HTTPIngressRuleValue struct {
-// the path are forwarded to the backend.
+// PathType represents the type of path referred to by a HTTPIngressPath.
+type PathType string
+
+const (
+	// PathTypeExact matches the URL path exactly.
+	PathTypeExact = PathType("Exact")
+
+	// PathTypePrefix matches based on a URL path prefix split by '/'. Matching is
+	// done on a path element by element basis. A path element refers is the
+	// list of labels in the path split by the '/' separator. A request is a
+	// match for path p if every p is an element-wise prefix of p of the
+	// request path. Note that if the last element of the path is a substring
+	// of the last element in request path, it is not a match (e.g. /foo/bar
+	// matches /foo/bar/baz, but does not match /foo/barbaz).
+	PathTypePrefix = PathType("Prefix")
+
+	// PathTypeImplementationSpecific matching is up to the IngressClass.
+	// Implementations are not required to support ImplementationSpecific
+	// matching.
+	PathTypeImplementationSpecific = PathType("ImplementationSpecific")
+)
+
+// HTTPIngressPath associates a path with a backend. Incoming urls matching the
+// path are forwarded to the backend.
 type HTTPIngressPath struct {
-	// Path is an extended POSIX regex as defined by IEEE Std 1003.1,
-	// (i.e this follows the egrep/unix syntax, not the perl syntax)
-	// matched against the path of an incoming request. Currently it can
-	// contain characters disallowed from the conventional "path"
-	// part of a URL as defined by RFC 3986. Paths must begin with
-	// a '/'. If unspecified, the path defaults to a catch all sending
-	// traffic to the backend.
+	// Path is an extended POSIX regex as defined by IEEE Std 1003.1, (i.e this
+	// follows the egrep/unix syntax, not the perl syntax) matched against the
+	// path of an incoming request. Currently it can contain characters
+	// disallowed from the conventional "path" part of a URL as defined by RFC
+	// 3986. Paths must begin with a '/'. If unspecified, the path defaults to a
+	// catch all sending traffic to the backend.
 	// +optional
 	Path string
 
+	// PathType determines the interpretation of the Path matching. PathType can
+	// be one of the following values:
+	// * Exact: Matches the URL path exactly. Must be supported by all
+	//   implementations.
+	// * Prefix: Matches based on a URL path prefix split by '/'. Matching is
+	//   done on a path element by element basis. A path element refers is the
+	//   list of labels in the path split by the '/' separator. A request is a
+	//   match for path p if every p is an element-wise prefix of p of the
+	//   request path. Note that if the last element of the path is a substring
+	//   of the last element in request path, it is not a match (e.g. /foo/bar
+	//   matches /foo/bar/baz, but does not match /foo/barbaz). Must be
+	//   supported by all implementations.
+	// * ImplementationSpecific: Interpretation of the Path matching is up to
+	//   the IngressClass. Implementations are not required to support
```

Comment:
If we make ImplementationSpecific the default, what are the implications of the requirement statement in this sentence?

Reply:
That's a great catch, thanks! I think we'll need to update the KEP to indicate that all implementations should support all path types, and if there is no obvious use for `ImplementationSpecific`, implementations can simply interpret that as a prefix.